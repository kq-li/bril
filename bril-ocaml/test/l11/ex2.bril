# ARGS: 42
@main(x: int): int {
  one: int = const 1;
  y: int = add x one;
  c: int = const 100;
  cond: bool = lt x c;
  br cond .true .false;
.true:
  z: int = call @f y;
  ret z;
.false:
  z: int = call @g y;
  ret z;
}

@f(x: int): int {
  one: int = const 1;
  x: int = sub x one;
  ret x;
}

@g(x: int): int {
  c: int = const -1;
  ret c;
}
