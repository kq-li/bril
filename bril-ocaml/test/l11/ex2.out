1c1
< total_dyn_inst: 9
---
> total_dyn_inst: 10
